modules/logging/efk/main.tf

```diff
@@ -1,75 +1,293 @@
+    var.module_depends_on
+  ]
 
-  set {
-    name  = "elasticsearch-curator.enabled"
-    value = var.elasticsearch-curator
-  }
+  name          = local.kibana_name
+  repository    = local.kibana_repository
+  chart         = local.kibana_chart
+  version       = var.kibana_chart_version
+  namespace     = local.namespace
+  recreate_pods = true
+  timeout       = 1200
+
+  dynamic "set" {
+    for_each = merge(local.kibana_conf)
 
-  set {
-    name  = "elasticsearch-curator.cronjob.failedJobsHistoryLimit"
-    value = var.failed_limit
+    content {
+      name  = set.key
+      value = set.value
+    }
   }
+}
 
-  set {
-    name  = "elasticsearch-curator.cronjob.successfulJobsHistoryLimit"
-    value = var.success_limit
+resource "local_file" "kibana" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content  = yamlencode(local.kibana_application)
+  filename = "${path.root}/${var.argocd.path}/${local.kibana_name}.yaml"
+}
+
+#Filebeat
+resource "helm_release" "filebeat" {
+  count = 1 - local.argocd_enabled
+
+  depends_on = [
+    var.module_depends_on
+  ]
+
+  name          = local.filebeat_name
+  repository    = local.filebeat_repository
+  chart         = local.filebeat_chart
+  version       = var.filebeat_chart_version
+  namespace     = local.namespace
+  recreate_pods = true
+  timeout       = 1200
+
+  dynamic "set" {
+    for_each = merge(local.filebeat_conf)
+
+    content {
+      name  = set.key
+      value = set.value
+    }
   }
+}
+
+resource "local_file" "filebeat" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content  = yamlencode(local.filebeat_application)
+  filename = "${path.root}/${var.argocd.path}/${local.filebeat_name}.yaml"
+}
 
-  set {
-    name  = "elasticsearch.data.persistence.size"
-    value = var.elasticDataSize
+
+locals {
+  #elasticsearch
+  elastic_name       = "elasticsearch"
+  elastic_repository = "https://helm.elastic.co/"
+  elastic_chart      = "elasticsearch"
+  elastic_conf       = merge(local.elastic_conf_defaults, var.elastic_conf)
+  #kibana
+  kibana_name       = "kibana"
+  kibana_repository = "https://helm.elastic.co/"
+  kibana_chart      = "kibana"
+  kibana_conf       = merge(local.kibana_conf_defaults, var.kibana_conf)
+  #filebeat
+  filebeat_name       = "filebeat"
+  filebeat_repository = "https://helm.elastic.co/"
+  filebeat_chart      = "filebeat"
+  filebeat_conf       = merge(local.filebeat_conf_defaults, var.filebeat_conf)
+  elastic_conf_defaults = {
+    "replicas" = var.elasticReplicas
+    "minimumMasterNodes" = var.elasticMinMasters
+    "volumeClaimTemplate.resources.requests.storage" = var.elasticDataSize
+  }
+  kibana_conf_defaults = {
+    "elasticsearchHosts" = "http://elasticsearch-master:9200"
+    "ingress.enabled"                                                   = "true"
+    "ingress.hosts[0]"                                                  = "kibana.${var.domains[0]}"
```

Comment:
in code bellow you define `var.domains` as a string variable 
``` hcl
variable "domains" {
  description = "domain name for ingress"
  default     = "example.com"
}
```
but use it as a non-zero length list of strings, maybe better to wrap all these options to the loop over `var.domains`?